### ISSUE TITLE
Complex file logic

### ISSUE THREAD

**Issue by oholiab** [aelsabbahy/goss#107]
Hey, I'm attempting to write a test that will pass if:
- the file doesn't exist
- the file does exist and contains "foo"

Is this possible? I've tried various combinations of putting "or" at different levels in my goss.yaml but I'm getting the impression that it doesn't work like that... a simple test case with:

    exists:
      or:
      - true
      - false

doesn't appear to work.

Is this something that is expected to work in the current version, and if not is it something that you consider to be within the scope of the project?

Thanks :)

**Comment by aelsabbahy** [comment]
Interesting use case. Unfortunately, support for this does not currently exist in `goss`.

It would be easy to add this feature if it's defined as: "If resource does not exist, skip ALL other tests."

If the feature requested is to do more complex logic like: "If the file is mode 0755, then ensure content X otherwise ensure content Y" that might be much more complex.

**Comment by oholiab** [comment]
I'd be fine with "skip if file does not exist" :)

Any more complex logic than that and I would agree that it's totally out of
scope and I should just write some Go ;)

**Comment by taurus-forever** [comment]
JFYI, we have moved all such logic on the layer before the goss execution.

We are using templates toolkit to build the server specific goss.json file (IP, services, server role, server configuration) and then trigger goss with plain goss.json, like:

```
> cat /etc/.../goss.json 
{
    "gossfile": {
        "01_running-processes.json": {},
        "01_hostname-and-interfaces.json": {},
        "02_disk-free.json": {},
        "03_files_availability.json": {},
        "03_no-cores.json": {},
        "03_open_ports.json": {},
        "04_template_files.json": {},
        "05_system_users.json": {},
        "06_system_packages.json": {},
        "07_system_service.json": {},
        "08_system_ports.json": {},
        "09_system_dns.json": {},
        "10_sites_ok.json": {},
        "99_dummy.json": {}
    }
}
```

You can find our repo here https://github.com/sipwise/system-tests ;-)
Hope it will help you a bit!

**Comment by aelsabbahy** [comment]
So, this will have to wait for the next release of goss. I had some time to work on it today, but prioritized fixing bugs over enhancements.

This change is a bit more work than I initially thought since currently tests have only a "Successful" boolean. I need to add a "Status" or "Result" which allows for success, failed, skipped and update the outputs to support that format.

**Comment by aelsabbahy** [comment]
Released in [v0.1.10](https://github.com/aelsabbahy/goss/releases/tag/v0.1.10)
